### PR TITLE
check APIVersions

### LINKED ISSUE #39672: kubectl error on nodes
Hi

I'm building a baremetal cluster on a  VPS. I have 2 masters and 3 nodes. Nodes are sceduling pods fine and the cluster as a whole is running fine.

The problem is when I try to run the following command on a node:

    kubectl get nodes

I see the following error

    $kubectl get nodes

    panic: runtime error: index out of range

    goroutine 1 [running]:
    panic(0x15e01e0, 0xc42000e0d0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
    k8s.io/kubernetes/pkg/client/typed/discovery.apiVersionsToAPIGroup(0xc4202cb860, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x20c9608, 0x0, 0x0, ...)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/client/typed/discovery/discovery_client.go:111 +0x222
    k8s.io/kubernetes/pkg/client/typed/discovery.(*DiscoveryClient).ServerGroups(0xc4204beee0, 0xc42025b8c0, 0x3c, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/client/typed/discovery/discovery_client.go:123 +0x5ee
    k8s.io/kubernetes/pkg/kubectl/cmd/util.(*CachedDiscoveryClient).ServerGroups(0xc420017fc0, 0x2, 0x0, 0x6)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/util/cached_discovery.go:117 +0x11a
    k8s.io/kubernetes/pkg/client/typed/discovery.GetAPIGroupResources(0x7f490f0e54f0, 0xc420017fc0, 0xc420017fc0, 0x7f490f0e54f0, 0xc420017fc0, 0xceae26, 0x1543f80)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/client/typed/discovery/restmapper.go:115 +0x5f
    k8s.io/kubernetes/pkg/kubectl/cmd/util.(*factory).UnstructuredObject(0xc4203d8860, 0x17840bd, 0x9, 0x0, 0x0, 0xc20dfe62cb76a168, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/util/factory.go:380 +0x9c
    k8s.io/kubernetes/pkg/kubectl/cmd.RunGet(0x205ed60, 0xc4203d8860, 0x2046000, 0xc420028010, 0x2046000, 0xc420028018, 0xc4204a4000, 0xc420415650, 0x1, 0x1, ...)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/get.go:152 +0x13d
    k8s.io/kubernetes/pkg/kubectl/cmd.NewCmdGet.func1(0xc4204a4000, 0xc420415650, 0x1, 0x1)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/get.go:105 +0x95
    k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute(0xc4204a4000, 0xc4204155d0, 0x1, 0x1, 0xc4204a4000, 0xc4204155d0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:603 +0x439
    k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4204e18c0, 0xc4204151d0, 0x1, 0x1)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:689 +0x367
    k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4204e18c0, 0xc4203d8860, 0x2045fc0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:648 +0x2b
    k8s.io/kubernetes/cmd/kubectl/app.Run(0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubectl/app/kubectl.go:38 +0xd5
    main.main()
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubectl/kubectl.go:26 +0x22

I'm running everything on 1.5.1 (kube-api/controller/manager/kubelet and kube-proxy)

I would expect this to return the nodes, since the node is connected to the api-server, right?

Thanks

### PR DESCRIPTION
**Pull request by du2016**
**What this PR does / why we need it**:
check Whether the APIVersions is empty.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes 
#39672

### CODE REVIEW

**Comment by resouer** [comment]
@k8s-bot ok to test

**Comment by resouer** [comment]
@k8s-bot ok to test

**Comment by du2016** [comment]
@k8s-bot unit test this

File: pkg/client/typed/discovery/discovery_client.go [hunk 1]
```diff
@@ -123,7 +123,7 @@ func (d *DiscoveryClient) ServerGroups() (apiGroupList *metav1.APIGroupList, err
 	v := &metav1.APIVersions{}
 	err = d.restClient.Get().AbsPath(d.LegacyPrefix).Do().Into(v)
 	apiGroup := metav1.APIGroup{}
-	if err == nil {
+	if err == nil && len(v.Versions) != 0 {
```
Comment (caesarxuchao) [inline pkg/client/typed/discovery/discovery_client.go]: We should check `len(v.Versions) != 0` before line 145 as well. We don't want to append an empty apiGroup to the apiGroupList.Groups.

This entire file is moved to staging/src/k8s.io/client-go/discovery/discovery_client.go

**Pull request closed without merging**